crates/jiter/src/py_lossless_float.rs

```diff
@@ -0,0 +1,71 @@
+use pyo3::exceptions::PyValueError;
+use pyo3::prelude::*;
+use pyo3::sync::GILOnceCell;
+use pyo3::types::PyType;
+
+use crate::Jiter;
+
+/// Represents a float from JSON, by holding the underlying bytes representing a float from JSON.
+#[derive(Debug, Clone)]
+#[pyclass]
+pub struct LosslessFloat(Vec<u8>);
+
+impl From<Vec<u8>> for LosslessFloat {
+    fn from(raw: Vec<u8>) -> Self {
+        Self(raw)
+    }
+}
+
+#[pymethods]
+impl LosslessFloat {
+    #[new]
+    fn new(raw: Vec<u8>) -> PyResult<Self> {
+        let s = Self(raw);
+        // check the string is valid by calling `as_float`
+        s.as_float()?;
+        Ok(s)
+    }
+
+    fn as_float(&self) -> PyResult<f64> {
+        let bytes = &self.0;
+        let mut jiter = Jiter::new(bytes, true);
+        let f = jiter
+            .next_float()
+            .map_err(|e| PyValueError::new_err(e.description(&jiter)))?;
+        jiter
+            .finish()
+            .map_err(|e| PyValueError::new_err(e.description(&jiter)))?;
```

Comment:
This is quite big brain 🧠. Is it worth considering a lower-level function to just try to parse the float as a number range and fail if it doesn't successfully consume the whole string?

Reply:
I think it's good as is.